crates/bevy_entity_inspector/src/ui/tree.rs

```diff
@@ -0,0 +1,600 @@
+                            _ => a.label.cmp(&b.label),
+                        }
+                    }
+                    _ => std::cmp::Ordering::Equal,
+                }
+            });
+
+            for root_id in &sorted_root_ids {
+                if let Some(root_node) = tree_state.nodes.get(root_id) {
+                    let node_entity = build_tree_node_recursive(
+                        &mut commands,
+                        root_node,
+                        &tree_state,
+                        &tree_config,
+                    );
+                    commands.entity(container_entity).add_child(node_entity);
+                }
+            }
+        }
+    }
+}
+
+/// Builds a tree view UI from the given state and configuration
+pub fn build_tree_view(
+    commands: &mut Commands,
+    _tree_state: &TreeState,
+    config: &TreeConfig,
+) -> Entity {
+    let frame = commands.spawn(tree_container(config)).id();
+
+    // We need to find the actual TreeContainer within the frame structure
+    // Since we can't easily access it here, we'll need to modify our approach
+
+    frame
+}
+
+/// Recursively builds tree nodes and their children using standard Bevy UI
+pub fn build_tree_node_recursive(
+    commands: &mut Commands,
+    node: &TreeNode,
+    tree_state: &TreeState,
+    config: &TreeConfig,
+) -> Entity {
+    let has_children = !node.children.is_empty();
+
+    // Create a container for the entire node (button + children)
+    let node_container = commands
+        .spawn(Node {
+            width: Val::Auto,
+            min_width: Val::Percent(100.0),
+            height: Val::Auto,
+            flex_direction: FlexDirection::Column,
+            ..default()
+        })
+        .id();
+
+    // Create the main clickable node row - make all nodes clickable for selection
+    let node_row = commands
+        .spawn((
+            Button, // All nodes are now buttons for selection
+            Node {
+                width: Val::Auto,
+                min_width: Val::Percent(100.0),
+                height: Val::Auto,
+                min_height: Val::Px(config.node_height),
+                flex_direction: FlexDirection::Row,
+                align_items: AlignItems::Center,
+                padding: UiRect::left(Val::Px(node.depth as f32 * config.indent_size)),
+                ..default()
+            },
+            TreeNodeWidget {
+                node_id: node.id.clone(),
+            },
+            BackgroundColor(if tree_state.selected_node.as_ref() == Some(&node.id) {
+                config.selected_color
+            } else if has_children {
+                Color::NONE
+            } else {
+                // Slightly different background for non-expandable components
+                Color::srgba(0.1, 0.1, 0.1, 0.3)
+            }),
+            BorderRadius::all(Val::Px(2.0)),
+        ))
+        .id();
+
+    // Add disclosure triangle or spacer for visual indication
```

Comment:
```suggestion
    // Add disclosure triangle for visual indication
```
I think this is clearer?